sdk/communication/azure-communication-chat/tests/test_chat_thread_client_e2e.py

```diff
@@ -149,12 +153,27 @@ def test_list_participants(self):
         if self.is_live:
             time.sleep(2)
 
-        chat_thread_participants = self.chat_thread_client.list_participants()
+        # add another participant
+        share_history_time = datetime.utcnow()
+        share_history_time = share_history_time.replace(tzinfo=TZ_UTC)
+        new_participant = ChatThreadParticipant(
+            user=self.new_user,
+            display_name='name',
+            share_history_time=share_history_time)
+
+        self.chat_thread_client.add_participant(new_participant)
+
+        # fetch list of participants
+        chat_thread_participants = self.chat_thread_client.list_participants(results_per_page=1, skip=1)
+
+        participant_count = 0
 
         for chat_thread_participant_page in chat_thread_participants.by_page():
             li = list(chat_thread_participant_page)
-            assert len(li) == 1
+            assert len(li) <= 1
```

Comment:
is it because, you need to `time.sleep()` here too?

Reply:
The tests have been updated. time.sleep() is now isolated within a single helper function now